lib/active_admin/filters/humanized.rb

```diff
@@ -9,8 +9,22 @@ def initialize(param)
         @value = param[1]
       end
 
+      def related_class
+        @related_class ||=
+            begin
+              @body.split('_id').first.gsub(' ', '').classify.constantize
+            rescue
+              nil
+            end
+      end
+
       def value
-        @value
+        @value =
+            begin
+              display_name(related_class.find(@value))
```

Comment:
I assume that I and people like me would not define this method often for any reason.
In fact it doesn't work at all if display_name won't define. I mean we won't get :title )
Are you sure?